components/meter/meter-linear.js

```diff
@@ -76,14 +77,19 @@ class MeterLinear extends LitElement  {
 		const percentage = this.max > 0 ? this.value / this.max * 100 : 0;
 		const primary = this.percent ? `${Math.floor(percentage)}%` : `${this.value}/${this.max}`;
 		const secondaryText = this._formatContext(this.text, this.value, this.max);
-		const secondaryTextElement = secondaryText ? html`<div class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
+		const secondaryTextElement = secondaryText ? html`<div id="d2l-meter-linear-secondary" class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
 		return html `
-			<div role="img" aria-roledescription="progress indicator" aria-label="${secondaryText}: ${primary}">
+			<div
+				role="progressBar"
+				aria-valuenow="${this.value}"
+				aria-valuemax="${this.max}"
+				aria-valuetext="${ifDefined(secondaryText)}"
+				aria-describedby="d2l-meter-linear-primary d2l-meter-linear-secondary">
 				<div class="d2l-meter-linear-full-bar">
 					<div class="d2l-meter-linear-inner-bar" style="width:${percentage}%;"></div>
```

Comment:
If it's still reading the percentage twice, you could put `role="presentation"` on this to hopefully hide it.

Reply:
On voice over, it went well. It would announce the `labelledby`, `percent`, progress indicator. After you stay on it. It will then announce "You are still on `DescribedBy`, progress indicator" So I think it is pretty good.

Reply:
Cool!